cmd/kubeadm/app/componentconfigs/defaults.go

```diff
@@ -97,25 +117,54 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 	// Enforce security-related kubelet options
```

Comment:
We have another 3 ifs above this line, that need just a warning.

Reply:
Thank @rosti, quick question about ClusterDNS,

ClusterDNS is an array, do we have to iterate (or sort and binary search) over this array to check if the recommended default value `10.96.0.10` is included? or just warn when ClusterDNS is not nil?

if iterate, this may introduce performance issue when user provide a large list.

So, could you please suggest a good solution?

Reply:
We should really check if `ClusterDNS` has a single value that is the default DNS IP. The whole thing should be on the lines of:

```golang
dnsIP, err := constants.GetDNSIP(internalcfg.Networking.ServiceSubnet)
if err != nil {
	dnsIP = kubeadmapiv1beta2.DefaultClusterDNSIP
}

if externalkubeletcfg.ClusterDNS == nil {
	externalkubeletcfg.ClusterDNS = []string{dnsIP}
} else if len(externalkubeletcfg.ClusterDNS) != 1 || externalkubeletcfg.ClusterDNS[0] != dnsIP {
	warnDefaultComponentConfig(...)
}
```
Hence no iteration is necessary.

Reply:
Thanks a lot, fixed.